### ISSUE TITLE
[Bug] Public Path not being fully added

### ISSUE THREAD

**Issue by gabrielboliveira** [bugsnag/webpack-bugsnag-plugins#11]
I've tried to add this to our webpack build script, however even if I use our full path with publicPath, it's not working since it resolves to the root url path. Let me explain:

Adding the plugin:

```js
new BugsnagSourceMapUploaderPlugin({
    apiKey: 'api-key-here',
    publicPath: 'http://example.com/some-path/',
    overwrite: true,
})
```

The files are uploaded successfully, however they have the `http://example.com/` base path, and not the full `http://example.com/some-path/`.

I've taken a quick look on the source code and seems to be here:

https://github.com/bugsnag/webpack-bugsnag-plugins/blob/b37fdd3657ba1286a3fffa1df4d899efca26b67a/source-map-uploader-plugin.js#L54-L58

Quickly testing on Node CLI:

```js
const url = require('url')

url.resolve('http://example.com/some-path/', '/main.js')
// 'http://example.com/main.js'

url.resolve('http://example.com/some-path/another-path/', '/main.js')
// 'http://example.com/main.js'

url.resolve('http://example.com/some-path/', 'main.js')
// 'http://example.com/some-path/main.js'

url.resolve('http://example.com/some-path/another-path/', 'main.js')
// 'http://example.com/some-path/another-path/main.js' 
```

It seems that the `source` variable have the `/` before the name. Perhaps removing the `/` before the file name or even adding a callback option as suggested on https://github.com/bugsnag/webpack-bugsnag-plugins/issues/8#issuecomment-385356627 may be an option.

By the way, we are using [laravel-mix](https://github.com/JeffreyWay/laravel-mix) as webpack wrapper.

**Comment by bengourley** [comment]
Hey @gabrielboliveira – thanks for the detailed report. I was able to reproduce this on a fresh laravel mix project.

I would argue that the leading slash in the chunk file name is a bug in laravel mix, however there might be good reason for its existence. It's a pretty easy workaround for us as far as I can see.

I've pushed a fix to [this branch](https://github.com/bugsnag/webpack-bugsnag-plugins/tree/bengourley/source-leading-slash) and published a prerelease version on npm: v1.2.1-0. You'll need to explicitly install that version:

```
npm i webpack-bugsnag-plugins@1.2.1-0
```

Please try it out and let me know if this resolves things for you. Cheers!

**Comment by gabrielboliveira** [comment]
Thanks for the quick fix @bengourley!

The laravel-mix is basically a Laravel helper package, so they must have a reason to add that backslash because of Laravel. We use it on our Vue project since it's already inside a Laravel project, and also it's a pretty handy package.

I just tested that version on our local, and it worked like a charm. That fix works 🎉

**Comment by bengourley** [comment]
This is now published as v1.2.1 on npm.
